src/UI/Component/Table/Factory.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/* Copyright (c) 2017 Nils Haagen <nhaageng@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Table;
+
+/**
+ * Table factory
+ */
+interface Factory {
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *       The Presentation Table lists some data from the system in a pleasant way.
+	 *       The user should be able to get a quick overview over records in a
+	 *       dataset as simple as possible; for this, only most relevant fields of
+	 *       a record are being displayed at first glance. The deeper exploration of
+	 *       data should be as easy: Rows can be expanded to show more extensive fields
+	 *       and additional data that is less important for the identification of a record.
+	 *
+	 *       The aim of the Presentation Table is to represent a dataset as an entirety
+	 *       rather than a list of single rows. The table focusses on exploring this set,
+	 *       while the purpose of this exploration is known and supported.
+	 *       A single record does not necessarily reference a persistent entity like an ilObject,
+	 *       but can be derived and composed from all kind of sources.
+	 *
+	 *   composition: >
+	 *       The Presentation Table consists of a title, a slot for View Controls
```

Comment:
What is the title of a dataset in a table? Is this always the first column? Please do not refer to title in case of a table but allow to identify the means how the next table can be processed and explain what column goes where

Reply:
The title is derived from the dataset via the aforementioned "row mapping closure" that needs to be supplied by the developer using that table. The same goes for all other fields in the records supplied as dataset for the table.

Maybe the word "title" is too preallocated with other meaning to use it in this context. Any suggestion for another word, preferably one where also is the notion of a "subtitle"?

I do not understand what you want me to identify.